script.js

```diff
@@ -358,12 +358,22 @@ closePopup();
 // form validation
 
 const form = document.getElementById('contact');
+const username = document.getElementById('name');
 const email = document.getElementById('email');
+const message = document.getElementById('textBody');
 const errormsg = document.getElementById('error');
 
 form.addEventListener('submit', (e) => {
   if (email.value.toLowerCase() !== email.value) {
     errormsg.style.display = 'block';
     e.preventDefault();
   }
-});
\ No newline at end of file
+
+  localStorage.setItem('name-input', username.value);
+  localStorage.setItem('email-input', email.value);
+  localStorage.setItem('msg-input', message.value);
+});
+
+username.value = localStorage.getItem('name-input');
+email.value = localStorage.getItem('email-input');
+message.value = localStorage.getItem('msg-input');
```

Comment:
[OPTIONAL] Please consider dividing your JavaScript logic into smaller pieces. Instead of a huge 400 lines file, you could have 5 files with 80 lines each 😉. This will make your code more maintainable and easier to debug. Also, it is a lot easier to understand what was your intentions, even for yourself, revisiting your own code after some time 😄.